src/cross_section/fillet2d.cpp

```diff
@@ -0,0 +1,946 @@
+        currentEdgeT = arc.t2;
+
+        tracingEList.push_back(currentEdgeIndex);
+        mapVV.push_back(rLoop.size());
+      }
+    }
+    newPoly.push_back(rLoop);
+  }
+
+#ifdef MANIFOLD_DEBUG
+  if (ManifoldParams().verbose) {
+    std::cout << "Result loop count:" << newPoly.size() << std::endl;
+  }
+#endif
+
+  return newPoly;
+}
+
+Polygons FilletImpl(const SimplePolygon& loop, double radius,
+                    int circularSegments) {
+  using namespace manifold;
+
+  ColliderInfo info{};
+  info.outerCollider = BuildCollider(loop, info.outerEdgeOld2NewVec);
+
+  // // Process inner loops
+  // info.innerCollider = std::vector<Collider>(polygons.size() - 1,
+  // Collider()); info.innerVec = std::vector<std::vector<edgeOld2New>>(
+  //     polygons.size() - 1, std::vector<edgeOld2New>());
+  // for (size_t i = 1; i != polygons.size(); i++) {
+  //   info.innerCollider[i] = BuildCollider(polygons[i], info.innerVec[i]);
+  // }
+
+  // Calc all arc that bridge 2 edge
+  auto arcConnection = CalculateFilletArc(loop, info, radius);
+
+  // Tracing along the arc
+  int n = circularSegments > 2 ? circularSegments
+                               : Quality::GetCircularSegments(radius);
+
+  auto result = Tracing(loop, arcConnection, n, radius);
+
+  return result;
+}
+
+}  // namespace
+
+namespace manifold {
+
+struct PathImpl {
+  PathImpl(const C2::PathsD paths_) : paths_(paths_) {}
+  operator const C2::PathsD&() const { return paths_; }
+  const C2::PathsD paths_;
+};
+
+std::vector<CrossSection> CrossSection::Fillet(double radius,
+                                               int circularSegments) const {
+  auto paths = this->GetPaths()->paths_;
+
+  Polygons outer, inner;
+  for (const auto& loop : paths) {
+    SimplePolygon polygon;
+    polygon.reserve(loop.size());
+
+    for (auto p : loop) {
+      polygon.push_back(v2_of_pd(p));
+    }
+
+    if (C2::Area(loop) > EPSILON)
+      outer.push_back(polygon);
+    else
+      inner.push_back(polygon);
+  }
+
+  std::vector<CrossSection> result;
+
+  for (const auto& loop : outer) {
+    auto r = FilletImpl(loop, radius, circularSegments);
+
+    for (const auto& ele : r) {
+      Polygons poly{ele};
+      poly.insert(poly.end(), inner.begin(), inner.end());
+      result.push_back(poly);
+    }
+  }
+
+  return result;
+}
+
+std::vector<CrossSection> CrossSection::Fillet(const SimplePolygon pts,
+                                               double radius,
+                                               int circularSegments) {
+  return Fillet(Polygons{pts}, radius, circularSegments);
+}
+
+std::vector<CrossSection> CrossSection::Fillet(const Polygons& polygons,
+                                               double radius,
+                                               int circularSegments) {
+  return CrossSection(polygons).Fillet(radius, circularSegments);
+}
+
+std::vector<CrossSection> Fillet(const std::vector<CrossSection>& crossSections,
```

Comment:
Why do we need vectors as input/output of this method? CrossSection is already a vector that contains any number of contours.